src/test/util/event.rs

```diff
@@ -120,7 +120,9 @@ pub struct EventHandler {
     command_events: EventQueue<CommandEvent>,
     pub pool_cleared_events: EventQueue<PoolClearedEvent>,
     sdam_events: EventQueue<SdamEvent>,
+    cmap_events: EventQueue<CmapEvent>,
```

Comment:
It looks like some of the new tests are pretty flaky. This might be due to the use of the `EventQueue` rather than the channel, since the former is more of a point-in-time measurement that can be easily susceptible to races. In general, I've been trying to move towards using channels for all the event assertions for this reason, since you can just listen to them (with a set timeout) until the expected number of events have come in.

Not sure if this is the actual problem here, but something to consider.

Reply:
I went a different direction here and introduced a sync point to ensure that all queued worker requests have been processed before checking the events.  Waiting on a channel seems like it has a few downsides that avoids, in that it'll still catch failure cases of too many events, and it doesn't require timeouts.

I wasn't able to reproduce these failures locally, though, so I'll have to wait for evergreen to see if it fixed the flakes; if not I'll shift to a channel.